src/server/saved_objects/client/saved_objects_client.js

```diff
@@ -0,0 +1,122 @@
+import Boom from 'boom';
+import { get } from 'lodash';
+
+import {
+  createFindQuery,
+  handleEsError,
+} from './lib';
+
+export class SavedObjectsClient {
+  constructor(kibanaIndex, callAdminCluster) {
+    this._kibanaIndex = kibanaIndex;
+    this._callAdminCluster = callAdminCluster;
+  }
+
+  async create(type, body = {}) {
+    const response = await this._withKibanaIndex('index', {
+      type,
+      body
+    });
+
+    return {
+      type: response._type,
+      id: response._id,
+      version: response._version,
+      attributes: body
+    };
+  }
+
+  async delete(type, id) {
+    const response = await this._withKibanaIndex('delete', {
+      type,
+      id,
+      refresh: 'wait_for'
+    });
+
+    if (get(response, 'found') === false) {
+      throw Boom.notFound();
+    }
+
+    return get(response, 'result') === 'deleted';
```

Comment:
Just curious, do you know why this wouldn't return `result: 'deleted'`? I think that returning a boolean here implies that consuming code should look something like this:

```js
try {
  const deleted = await savedObjects.delete('dashboard', 'xyz')
  if (!deleted) {
    throw new Error('not sure why, but the object was not deleted??');
  }
} catch (err) {
  // handle error
}
```

That said, I'm pretty sure that: the `result` attribute on the response is only there so that responses for create/updated/etc have a similar shape, we don't need to check it, and this method should throw if the document is not deleted for whatever reason (otherwise return nothing)